src/architecture/clients.md

```diff
@@ -68,17 +74,37 @@ implemented in Exonum light client.
   that links the response to the block header.
 3. The client, on receiving the response from the blockchain, *verifies the structure*
   and *validates cryptographic proofs* for the response.
-4. The verification procedure includes *checking whether a returned response
+  The verification procedure includes *checking whether a returned response
   is stale*. This is accomplished by calculating the median of timestamps recorded
   in `Precommit`s and comparing it against the local time of the client.
   If the median time in `Precommit`s is too far in the past, the response
   is considered stale, and its verification fails.
-5. The result of checks is shown in the user interface
+4. The result of checks is shown in the user interface
 
 !!! note
     In the case user authentication is needed (for example, for data
     access management), requests can be *digitally signed*.
 
+An example of a cryptographic proof:
+
+![Cryptographic proof](../images/proof.png)
+
+In this schema, the data is in the Merkle tree (or its variants) with the rest of the data of the corresponding type.
+
+The root hash of this Merkle tree is called **state hash**,
+which means that it is a hash of all data in the blockchain.
+This hash is written in the block, which is signed by the validators.
+
+To prove the existence of data in the blockchain, it is enough to first check the Merkle tree,
```

Comment:
This and the following paragraph seem unnecessary to me; this paragraph repeats the verification procedure above, and the following one is only marginally related to the light client. Consider removing.